SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayNonce.java

```diff
@@ -22,9 +23,14 @@ static SamsungPayNonce fromJSON(JSONObject inputJson) throws JSONException {
         JSONObject data = new JSONObject(inputJson.getString("data"));
         JSONObject braintreeData = new JSONObject(data.getString("data"));
 
-        JSONObject paymentMethod = braintreeData
-                .getJSONObject("tokenizeSamsungPayCard")
-                .getJSONObject("paymentMethod");
+        JSONObject tokenizeSamsungPayResponse =
+            braintreeData.getJSONObject("tokenizeSamsungPayCard");
+
+        JSONObject paymentMethod = tokenizeSamsungPayResponse.optJSONObject("singleUseToken");
```

Comment:
Should the fallback instead be `singleUseToken`, especially if `paymentMethod` is the new name in the GW?

Reply:
Yeah that's a good call I can flip the logic.